File: paul-beatrice-student-code-main/module-3/week-8/lecture-final/client-vue/src/services/ProductService.js
import axios from 'axios';

const productService = {
    getAllProducts() {
        
      return axios.get('/products');
    },
  
    getProductById(id) {
      return axios.get(`/products/${id}`);
    },
  
    createProduct(product) {
      return axios.post('/products', product);
    },
  
    updateProduct(id, product) {
      return axios.put(`/products/${id}`, product);
    },
  
    deleteProduct(id) {
      return axios.delete(`/products/${id}`);
    },
  };
  
  export default productService;

File: paul-beatrice-student-code-main/module-3/week-8/lecture-final/server-java/src/main/java/com/techelevator/service/ProductService.java
package com.techelevator.service;
import com.techelevator.model.Product;
import java.util.List;
public interface ProductService {

    Product createProduct(Product product);
    Product getProductById(int id);
    List<Product> getAllProducts();
    Product updateProduct(Product product);
    void deleteProduct(int id);
}
